.github/workflows/message_gray_integration_test.yml

```diff
@@ -0,0 +1,120 @@
+    steps:
+      - uses: actions/checkout@v3
+      - name: cache local cse
+        uses: actions/cache@v3
+        with:
+          path: Local-CSE-2.1.3-linux-amd64.zip
+          key: ${{ runner.os }}-local-cse
+          restore-keys: |
+            ${{ runner.os }}-local-cse
+      - name: download cse
+        run: |
+          export ROOT_PATH=$(pwd)
+          bash ./sermant-integration-tests/scripts/tryDownloadMidware.sh cse
+      - name: cache rocketmq
+        uses: actions/cache@v3
+        with:
+          path: rocketmq-all-5.1.4-bin-release.zip
+          key: ${{ runner.os }}-rocketmq-all-5.1.4-bin-release
+          restore-keys: |
+            ${{ runner.os }}-rocketmq-all-5.1.4-bin-release
+      - name: download rocketmq
+        run: |
+          export ROOT_PATH=$(pwd)
+          bash ./sermant-integration-tests/scripts/tryDownloadMidware.sh rocketmq514
+  build-agent-and-cache:
+    name: build agent and cache
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 8
+        uses: actions/setup-java@v3
+        with:
+          java-version: '8'
+          distribution: 'adopt'
+          cache: maven
+      - name: cache agent
+        uses: actions/cache@v3
+        with:
+          path: sermant-agent-*/
+          key: ${{ runner.os }}-agent-${{ github.run_id }}
+      - name: package agent
+        run: |
+          sed -i '/sermant-backend/d' pom.xml
+          sed -i '/sermant-integration-tests/d' pom.xml
+          sed -i '/sermant-injector/d' pom.xml
+          sed -i '/sermant-flowcontrol/d' sermant-plugins/pom.xml
+          sed -i '/sermant-database-write-prohibition/d' sermant-plugins/pom.xml
+          sed -i '/sermant-spring-beans-deal/d' sermant-plugins/pom.xml
+          sed -i '/sermant-service-removal/d' sermant-plugins/pom.xml
+          sed -i '/sermant-service-visibility/d' sermant-plugins/pom.xml
+          sed -i '/sermant-monitor/d' sermant-plugins/pom.xml
+          sed -i '/sermant-mq-consume-prohibition/d' sermant-plugins/pom.xml
+          sed -i '/sermant-springboot-registry/d' sermant-plugins/pom.xml
+          sed -i '/sermant-service-registry/d' sermant-plugins/pom.xml
+          sed -i '/sermant-dynamic-config/d' sermant-plugins/pom.xml
+          sed -i '/sermant-router/d' sermant-plugins/pom.xml
+          sed -i '/sermant-loadbalancer/d' sermant-plugins/pom.xml
+          mvn package -DskipTests -Ptest --file pom.xml
+  test-for-grayscale-rocketmq:
+    name: Test for grayscale rocketmq
+    runs-on: ubuntu-latest
+    if: needs.set-execution-conditions.outputs.enableMqGrayscaleRocketMqAction=='true'
+    needs: [set-execution-conditions, build-agent-and-cache, download-midwares-and-cache]
+    strategy:
+      matrix:
+        test-model: [ 'PLUGIN_ENABLED_FALSE','CONSUMER_BASE_ONLY','CONSUMER_BASE_GRAY' ]
+        rocketMqClientVersion: ["4.8.0", "4.9.0", "4.9.1", "4.9.2", "4.9.3", "4.9.4", "4.9.5", "4.9.6", "4.9.7", "4.9.8", "5.0.0"]
```

Comment:
5.1.x version?

Reply:
SubscriptionData class path changed from 5.1.x，and bug fixed later.